share/spice/octopart/octopart.js

```diff
@@ -1,90 +1,60 @@
-    //     isRelevant = DDG.isRelevant(iter.item.mpn, skip_words, 4, true) || 
-    //         DDG.isRelevant(iter.item.manufacturer.displayname, skip_words, 4, true);
-
-    //     // Check if we have images.
-    //     if(iter.item.images.length > 0 && isRelevant) {
-    //         results.push(iter);
-    //     // If an image doesn't exist, add a different image.
-    //     } else if(isRelevant) {
-    //         iter.item.images.push({
-    //             url_90px: "http://n1.octostatic.com/o3web/detail/images/camera-icon.png"
-    //         });
-    //         results.push(iter);
-    //     }
-    // }
-
-    // if(results.length === 0) {
-    //     return;
-    // }
-
-    var isMobile = $('.is-mobile').length;
-
-    Spice.add({
-        id: 'octopart',
-        name: 'Parts',
-        data: api_result.results,
-        signal: 'high',
- 
-        meta: {
-            itemType: 'Parts',
-            sourceName: 'Octopart',
-			sourceUrl : 'http://octopart.com/partsearch#search/requestData&q=' + api_result.request.q
-        },
-        normalize: function(item) {
-            var img = DDG.getProperty(item, "item.images.0.url_90px");
-            var desc_length = 60;
-            var description = (isMobile ? 
-                Handlebars.helpers.ellipsis(DDG.strip_html(item.item.short_description), desc_length) :
-                DDG.strip_html(item.item.short_description));
-
-            return {
-                brand: item.item.manufacturer.displayname,
-                price: item.item.avg_price[1] + ' ' + item.item.avg_price[0].toFixed(2),
-                img: img,
-                img_m: img,
-				url: item.item.detail_url,
-                title: DDG.strip_html(item.item.mpn).toUpperCase(),
-                heading: DDG.strip_html(item.item.mpn).toUpperCase(),
-                description: description,
-                datasheet: item.item.datasheets[0].url
-            };
-        },
+        if(!api_result || !api_result.results || api_result.results.length === 0) {
+            return Spice.failed('octopart');
+        }
 
-        templates: {
-            group: 'products',
-            options: {
-                rating: false,
-                buy: Spice.octopart.buy
+        Spice.add({
+            id: 'octopart',
+            name: 'Parts',
+            data: api_result.results,
+            signal: 'high',
+            meta: {
+                itemType: 'Parts',
+                sourceName: 'Octopart',
+                sourceUrl : 'http://octopart.com/search?q=' + api_result.request.q
+            },
+            normalize: function(item) {
+
+                item = item.item; // our item object is wrapped in an "item" property
+
+                var img_url = item.images[0].url;
+
+                return {
+                    brand: item.manufacturer.displayname,
+                    price: item.avg_price[1] + ' $' + item.avg_price[0].toFixed(2),
```

Comment:
Looks like this is the old API. Looking at the request, there isn't an `avg_price` but an `avg_price_v2`